homeassistant/components/honeywell/sensor.py

```diff
@@ -0,0 +1,66 @@
+"""Support for Honeywell (US) Total Connect Comfort sensors."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import SensorEntity, SensorStateClass
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Honeywell thermostat."""
+    data = hass.data[DOMAIN][config.entry_id]
+    sensors = []
+
+    for device in data.devices.values():
+        if device.outdoor_temperature is not None:
+            sensors.append(HoneywellOutdoorTemperatureSensor(device))
+        if device.outdoor_humidity is not None:
+            sensors.append(HoneywellOutdoorHumiditySensor(device))
```

Comment:
You could implement `SensorEntityDescription`s instead. It would produce less code and better readable code :)

Reply:
I think I would still require these `if` statements to determine which `device_class` should be used though, no? I can see `SensorEntityDescription` being useful to reduce the number of Sensor classes I create (from 2 to 1) but then I would need to add `if` statements for figuring out the unit and value. Is using `if` statements preferable to separate Sensor classes? I was told to use classes in the last PR haha. I'm happy with either way TBH.

Reply:
Entity descriptions remove the need to have 2 sensor classes in the first place.

Reply:
Sounds good. If that's the preferred way I'll make the swap to `SensorEntityDescriptuon` and a couple `if` statements.